### PR TITLE
Feat: Expires bufferView after resizing it's source buffer.

### PR DESCRIPTION
**Pull request by bluesky013**
Re: #

### Changelog

* Expires bufferView after resizing it's source buffer.

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by bluesky013** [comment]
@cocos-robot run test cases

File: native/cocos/renderer/gfx-validator/DescriptorSetValidator.cpp [hunk 1]
```diff
@@ -116,7 +116,9 @@ void DescriptorSetValidator::updateReferenceStamp() {
 
 void DescriptorSetValidator::bindBuffer(uint32_t binding, Buffer *buffer, uint32_t index) {
     CC_ASSERT(isInited());
-    CC_ASSERT(buffer && static_cast<BufferValidator *>(buffer)->isInited());
+    auto *vBuffer = static_cast<BufferValidator *>(buffer);
+    CC_ASSERT(buffer && vBuffer->isInited());
+    CC_ASSERT(vBuffer->isValid() && "Buffer View Expired");
```
Comment (minggo) [inline native/cocos/renderer/gfx-validator/DescriptorSetValidator.cpp:121]: "Buffer View Expired" is not needed, or just treat it as comment?

**Review by minggo — APPROVED** [review]

File: native/cocos/renderer/gfx-validator/DescriptorSetValidator.cpp (hunk 1 above)
Comment (bluesky013) [reply native/cocos/renderer/gfx-validator/DescriptorSetValidator.cpp:121]: Assert shows the condition if failed. Just a Hint.

**Review by star-e — APPROVED** [review]

**Comment by bluesky013** [comment]
@cocos-robot run test cases

**Pull request merged**